### ISSUE TITLE
Internet Speed Xpenology VM on a Synology VMM very slow!

### ISSUE THREAD

**Issue by BobWs** [fbelavenuto/arpl#213]
Hi,
As I understand from the Xpenology Forum, it is possible to install your software on VMWare using the ***.vmdk file.
But do you know if it is possible to install it on a real Synology via the VMM (Virtual Machine Manager)? 
Has anyone tried it successfully?

**Comment by fbelavenuto** [comment]
Please try with another network models. I don't know what models are available, if there is more, test it, ARPL puts several modules for the DSM and the load is automatic, so changing the model of the emulated network board should work without having to redo the loader.

**Comment by BobWs** [comment]
> Please try with another network models. I don't know what models are available, if there is more, test it, ARPL puts several modules for the DSM and the load is automatic, so changing the model of the emulated network board should work without having to redo the loader.

I can choose between these in the Synology VMM (see screenshot)
![Screen_Shot](https://user-images.githubusercontent.com/5485820/193812497-1fdc81ee-149b-4a79-975a-5480959b2c54.png)

I can confirm that e1000 is working, also tried the virtio and rlt8139 but that wasn't working. Getting a IP error. (see screenshot).
![Screen_Shot 1](https://user-images.githubusercontent.com/5485820/193813632-09f2e322-70ef-4f7f-bbc7-6968809ec40f.png)

**Comment by fbelavenuto** [comment]
Working now?

**Comment by BobWs** [comment]
> Working now?

Only the e1000 is working but then the internet speed is slower than on the host. So I guess I should say no it isn’t working properly.

Overall everything else is working fine.

**Comment by fbelavenuto** [comment]
OK. There are modules for virtio, it was supposed to work!

**Comment by BobWs** [comment]
> OK. There are modules for virtio, it was supposed to work!

Yes, I choose the module Virto but it didn’t work. Maybe the virtio driver that Synology is using is different from the modules you have?🤷‍♂️

**Comment by BobWs** [comment]
Here are some speed-tests I have done between Xpenology and a real Synology Virtual DSM installed in a real DS920+ with VMM.

As you can see the problem is that the speed on the XPE is very low compared to the real Synology VDSM when OpenVPN connection in DSM is enabled.

The test is done with VPN because that is what I'm using my VDSM for, it service as a VPN gateway for my home LAN.
I'm in need for a second VPN Gateway and thought to use the Xpenology for this because of paid licensing for Synology VDSM. I get only 1 free License when purchased my DS920+. 

But unfortunately the speed is to slow on Xpenology when VPN is active.
Is there anything that can be done to solve this? I don't understand why this is happening!

**Speedtest on VDSM with VPN Connection ON**
![IMG_0140](https://user-images.githubusercontent.com/5485820/195989982-6a9c6a4e-47ec-4d25-ba43-e9033010a627.png)


**Speedtest on Xpenology with VPN Connection ON**
![IMG_0141](https://user-images.githubusercontent.com/5485820/195990107-40a9eca7-b47f-4a86-bcb6-47907b2f7b96.png)


**Speedtest on Xpenology with VPN Connection OFF**
![IMG_0142](https://user-images.githubusercontent.com/5485820/195990139-4dc968a2-8457-437a-a984-617f77603807.png)

**Comment by fbelavenuto** [comment]
Please update ARPL, reboot, update modules, addons and lkm, rebuild the loader and test it.

**Comment by BobWs** [comment]
> Please update ARPL, reboot, update modules, addons and lkm, rebuild the loader and test it.

I've installed a clean version with the latest beta12 and the internet speed with e1000 network-card has improved!
**Speedtest with VPN on:**
![Screen_Shot](https://user-images.githubusercontent.com/5485820/213731310-8fe29c3f-681c-4566-86fd-69e9507a3551.png)


But when choosing the virtIO it isn't working. 
![Screen_Shot 1](https://user-images.githubusercontent.com/5485820/213731364-2ebca1f7-eb1c-414f-a633-533431e962ac.png)
_When choosing this setting it isn't working I'm getting a error that there is no IP and cannot create the loader. I have to choose the e1000 card._

**Comment by traeu** [comment]
Maybe I got a similar problem: I'm using TrueNAS 13.0 U3.1 which uses Bhyve as hypervisor. When I use the e1000 network adapter, everything works as expected. When I use virtio, the loader says "can't find device eth0" and gets no IP.
It's not really a problem for me, I only use this VM for experiments and e1000 works fine for that.
Just wanted to report that virtio is also not working for me.

**Comment by BobWs** [comment]
> "can't find device eth0" and gets no IP.

Yes same error message when choosing virtio on a real Synology Virtual Machine Manager.

**Comment by fbelavenuto** [comment]
It seems that there is something different in the "virtio" of VMM and Truenas, because in qemu it works perfect.

**Comment by BobWs** [comment]
> It seems that there is something different in the "virtio" of VMM and Truenas, because in qemu it works perfect.

Do you think that it is possible for you to fix? The official VDSM for VMM from Synology works fine with virtio. Maybe that could be a reference to fix?
